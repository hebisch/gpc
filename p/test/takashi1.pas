program exercise28(input, output);
const smax  = 8;
      smax2 = 9;
      n     = 10;
type str    = packed array[1..smax] of char;
     str2   = packed array[1..smax2] of char;
var i, j: integer;
    s: array[1..n] of str; tmp: str;
    s2: array[1..n] of str2;
begin
{ the following part uses the 8 chars packed array}

   s[1]:='abababab';
   s[2]:='babababa';
   s[3]:='12121212';
   s[4]:='21212121';
   s[5]:='cdcdcdcd';
   s[6]:='dcdcdcdc';
   s[7]:='56565656';
   s[8]:='65656565';
   s[9]:='bcbcbcbc';
   s[10]:='cbcbcbcb';

   for i := 1 to n - 1 do
                for j := n - 1 downto i do
                        if s[j] > s[j + 1] then
                        begin
                                tmp := s[j];
                                s[j] := s[j + 1];
                                s[j + 1] := tmp
                        end;
        for i := 1 to n do
           writeln('s[', i:2, '] = ', s[i]);

{ the following part uses the 9 chars packed array}

   s2[1]:='abababab';
   s2[2]:='babababa';
   s2[3]:='12121212';
   s2[4]:='21212121';
   s2[5]:='cdcdcdcd';
   s2[6]:='dcdcdcdc';
   s2[7]:='56565656';
   s2[8]:='65656565';
   s2[9]:='bcbcbcbc';
   s2[10]:='cbcbcbcb';

   for i := 1 to n - 1 do
                for j := n - 1 downto i do
                        if s2[j] > s2[j + 1] then
                        begin
                                tmp := s2[j];
                                s2[j] := s2[j + 1];
                                s2[j + 1] := tmp
                        end;
        for i := 1 to n do
                writeln('s2[', i:2, '] = ', s2[i])
end.
