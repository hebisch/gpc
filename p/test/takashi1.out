s[ 1] = 12121212
s[ 2] = 21212121
s[ 3] = 56565656
s[ 4] = 65656565
s[ 5] = abababab
s[ 6] = babababa
s[ 7] = bcbcbcbc
s[ 8] = cbcbcbcb
s[ 9] = cdcdcdcd
s[10] = dcdcdcdc
s2[ 1] = 12121212 
s2[ 2] = 21212121 
s2[ 3] = 56565656 
s2[ 4] = 65656565 
s2[ 5] = abababab 
s2[ 6] = babababa 
s2[ 7] = bcbcbcbc 
s2[ 8] = cbcbcbcb 
s2[ 9] = cdcdcdcd 
s2[10] = dcdcdcdc 
